tests/translator/output/aws-cn/graphqlapi_ddb_datasource_reference.json

```diff
@@ -0,0 +1,327 @@
+{
+  "Resources": {
+    "GetTodoResolver": {
+      "Description": "Don't do anything, just reference",
+      "Properties": {
+        "ApiId": {
+          "Fn::GetAtt": [
+            "SuperCoolAPI",
+            "ApiId"
+          ]
+        },
+        "DataSourceName": {
+          "Fn::GetAtt": [
+            "SuperCoolAPIAnotherDataSource",
+            "Name"
+          ]
+        },
+        "FieldName": "getNote",
+        "RequestMappingTemplate": "{\n  \"version\": \"2017-02-28\",\n  \"operation\": \"GetItem\",\n  \"key\": {\n    \"NoteId\": $util.dynamodb.toDynamoDBJson($context.arguments.id)\n  }\n}\n",
+        "ResponseMappingTemplate": "$util.toJson($context.result)",
+        "TypeName": "Query"
+      },
+      "Type": "AWS::AppSync::Resolver"
+    },
+    "SuperCoolAPI": {
+      "Properties": {
+        "AuthenticationType": "AWS_IAM",
+        "LogConfig": {
+          "CloudWatchLogsRoleArn": {
+            "Fn::GetAtt": [
+              "SuperCoolAPICloudWatchRole",
+              "Arn"
+            ]
+          },
+          "FieldLogLevel": "ALL"
+        },
+        "Name": "SomeApi",
+        "Tags": [
+          {
+            "Key": "key1",
+            "Value": "value1"
+          },
+          {
+            "Key": "key2",
+            "Value": "value2"
+          }
+        ],
+        "XrayEnabled": true
+      },
+      "Type": "AWS::AppSync::GraphQLApi"
+    },
+    "SuperCoolAPIAnotherDataSource": {
```

Comment:
It looks like logical ID is `<api-id><dynamodb-datasource-name>` and currently data sources are under `DynamoDBDataSources`. When we add another category of data sources, and customer same name, it'll cause deployment failure due to logical ID clash. Is that expected?

Reply:
A good point, but not the one to address in this PR @pradhapanr